Tendency application update

grantfirl opened the pull request:
<!-- INSTRUCTIONS: 
- PLEASE READ/FOLLOW THE DIRECTIONS IN EACH SECTION
- Complete the 'Commit Queue Requirements' below
- Please use github markup as much as possible (https://docs.github.com/en/get-started/writing-on-github)
- Please leave your PR in a draft state until all underlying work is completed.
-->
## Commit Queue Requirements:
<!--
- Please complete the items that follow this.
- Please "check off" completed items. Use [X] for a filled in checkbox or leave it [ ] for an empty checkbox
- Your PR will not be considered until all requirements are met.
- THIS IS YOUR RESPONSIBILITY
 -->
- [ ] Fill out all sections of this template.
- [ ] All sub component pull requests have been reviewed by their code managers.
- [ ] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [ ] Commit 'test_changes.list' from previous step
---
## Description:
<!--
Please provide a detailed verbose description of what this PR does
-->


### Commit Message:
<!--
Please provide concise information for The UFS-WM and/or each sub-component:
Please delete what is not needed.
-->
```
* UFSWM - 
  * AQM - 
  * CDEPS - 
  * CICE - 
  * CMEPS - 
  * CMakeModules - 
  * FV3 - 
    * ccpp-physics - 
    * atmos_cubed_sphere - 
  * GOCART - 
  * HYCOM - 
  * MOM6 - 
  * NOAHMP - 
  * WW3 - 
  * fire_behavior
  * stochastic_physics - 
```

### Priority:
<!--
Please provide the priority you would prefer this pull request to have.
* Critical Bugfix: Model is wrong.
* High: Time-sensitive project.
* Normal.
Please delete the ones that are not applicable
-->
* Critical Bugfix: Reason
* High: Reason
* Normal

## Git Tracking
### UFSWM:
<!--
Please add the UFS-WM github issue here if there is one
Please delete the one that is not applicable.
-->
* Closes #
* None

### Sub component Pull Requests:
<!--
Please provide a list of sub-components involved with this pull request.
Please provide links to the sub-component pull requests as shown below.
Please delete what is not needed.
Example:
* FV3: NOAA-EMC/fv3atm#734
  * ccpp-physics: ufs-community/ccpp-physics#33
* WW3: NOAA-EMC/WW3#321
-->
* AQM:
* CDEPS:
* CICE:
* CMEPS:
* CMakeModules:
* FV3:
  * ccpp-physics:
  * atmos_cubed_sphere:
* GOCART:
* HYCOM:
* MOM6:
* NOAHMP:
* WW3:
* fire_behavior:
* stochastic_physics:
* None

### UFSWM Blocking Dependencies:
<!--
If there are any UFSWM PR's that are needed to be completed before this one, please add links
to them here
Please delete what is not needed.
-->
* Blocked by #
* None

### Documentation:
<!--
Indicate what documentation, if any, is needed for this PR and who will add it (if applicable).
Please delete what is not needed.
-->
* This PR requires a documentation update, and the WM User's Guide has been updated based on the changes in this PR.
* This PR requires a documentation update, and a WM issue has been opened to track the need for a documentation update; a person responsible for submitting the update has been assigned to the issue (link issue). 
* No documentation update is required for this PR (please explain).

---
## Changes
### Regression Test Changes (Please commit test_changes.list):
<!--
Please let us know if this PR creates new baselines, changes baselines or not.
Please delete what is not needed.
Please make sure you have properly submitted test_changes.list
-->
* PR Adds New Tests/Baselines.
* PR Updates/Changes Baselines.
* No Baseline Changes.

### Input data Changes:
<!--
If there are any changes to input-data for a test, please provide information here.
Please delete what is not needed.
-->
* None.
* New input data.
* Updated input data.

### Library Changes/Upgrades:
<!-- Library updates take time. Please provide library and version information here.
** SPECIAL INSTRUCTIONS **
If this PR needs updates to libraries please make sure to accomplish the following tasks:
- Create separate issue in (https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
- Add issue link from JCSDA/spack-stack following this item <!-- for example: "* JCSDA/spack-stack#1757"

Please delete what is not needed.
-->
* Required
  * Library names w/versions:
  * Git Stack Issue (JCSDA/spack-stack#)
* No Updates
  
---
<!-- STOP!!! THE FOLLOWING IS FOR CODE MANAGERS ONLY. PLEASE DO NOT FILL OUT -->
## Testing Log:
- RDHPCS
  - [ ] Hera
  - [ ] Orion
  - [ ] Hercules
  - [ ] GaeaC6
  - [ ] Derecho
  - [ ] Ursa
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

gspetro-NOAA commented:
@grantfirl Are there additional changes expected for this PR (other than conflict resolution), or should we be reviewing it for Commit Queue readiness?

grantfirl commented:
> @grantfirl Are there additional changes expected for this PR (other than conflict resolution), or should we be reviewing it for Commit Queue readiness?

@gspetro-NOAA EMC wants to wait until after GFSv17 operational implementation to review/merge this, so it won't be ready for the commit queue for a while.

gspetro-NOAA commented:
@grantfirl Just checking in--is there a status update on this PR? It's been in draft for a while, so I just wanted to confirm that you're still planning to work on it.

grantfirl commented:
> @grantfirl Just checking in--is there a status update on this PR? It's been in draft for a while, so I just wanted to confirm that you're still planning to work on it.

Yes, indeed. I've been keeping this branch relatively up-to-date, but there are a few failing RTs that need to be fixed before removing the draft label. Even then, I think that EMC wants to do some more thorough testing to make sure that the result differences are below a threshold before merging.

gspetro-NOAA commented:
@grantfirl It looks like this PR includes a CMEPS update, but there is no corresponding CMEPS PR listed. Is this a mistake?

grantfirl commented:
> @grantfirl It looks like this PR includes a CMEPS update, but there is no corresponding CMEPS PR listed. Is this a mistake?

I forgot to open a PR in CMEPS. There should be one due to CCPP metadata changes needed there.

grantfirl commented:
> > @grantfirl It looks like this PR includes a CMEPS update, but there is no corresponding CMEPS PR listed. Is this a mistake?
> 
> I forgot to open a PR in CMEPS. There should be one due to CCPP metadata changes needed there.

@gspetro-NOAA Fixed.

gspetro-NOAA commented:
@grantfirl I saw that the CCPP PR has gotten approval from RhaeSung. I imagine you're planning to run the RT suite again, but will this PR likely be ready after that, or is there more work to be done?

grantfirl commented:
> @grantfirl I saw that the CCPP PR has gotten approval from RhaeSung. I imagine you're planning to run the RT suite again, but will this PR likely be ready after that, or is there more work to be done?

Yes, I'm going to run the RT suite again on Ursa. If successful, then it should be ready! I'm just doing final changes to update to the latest develop branches before kicking off RTs.

grantfirl commented:
@gspetro-NOAA @rhaesung I re-ran RTs on ursa and the results look to be the same, so I think that this is ready.

@gspetro-NOAA All tests using the RAS convection scheme fail to run, but this is expected. I have orders to remove this scheme from CCPP in a followup PR, so should we remove RAS tests from rt.conf in this PR, knowing that they fail, but it is of no consequence?

gspetro-NOAA commented:
> @gspetro-NOAA @rhaesung I re-ran RTs on ursa and the results look to be the same, so I think that this is ready.
> 
> @gspetro-NOAA All tests using the RAS convection scheme fail to run, but this is expected. I have orders to remove this scheme from CCPP in a followup PR, so should we remove RAS tests from rt.conf in this PR, knowing that they fail, but it is of no consequence?

I think it's best to remove those tests as part of this PR, and @dpsarmie agrees. It will mess up RT testing if we have a handful of tests that are failing but are supposed to fail (easy to miss others that are failing but not supposed to fail). Have project leads for those tests been notified? Just want to be sure it doesn't take anyone by surprise, but it sounds like there were already plans to remove them. @BrianCurtis-NOAA @NickSzapiro-NOAA Let us know if you have different thoughts on the matter.

grantfirl commented:
> > @gspetro-NOAA @rhaesung I re-ran RTs on ursa and the results look to be the same, so I think that this is ready.
> > @gspetro-NOAA All tests using the RAS convection scheme fail to run, but this is expected. I have orders to remove this scheme from CCPP in a followup PR, so should we remove RAS tests from rt.conf in this PR, knowing that they fail, but it is of no consequence?
> 
> I think it's best to remove those tests as part of this PR, and @dpsarmie agrees. It will mess up RT testing if we have a handful of tests that are failing but are supposed to fail (easy to miss others that are failing but not supposed to fail). Have project leads for those tests been notified? Just want to be sure it doesn't take anyone by surprise, but it sounds like there were already plans to remove them. @BrianCurtis-NOAA @NickSzapiro-NOAA Let us know if you have different thoughts on the matter.

Other than @yangfanglin, who is aware and OK'd the removal of RAS from the CCPP, I'm not sure what other leads would need to be notified. Is there any kind of list of projects/people that require certain RTs to exist in rt.conf? I'm not aware of it if so.

gspetro-NOAA commented:
> Other than @yangfanglin, who is aware and OK'd the removal of RAS from the CCPP, I'm not sure what other leads would need to be notified. Is there any kind of list of projects/people that require certain RTs to exist in rt.conf? I'm not aware of it if so.

Still no 😕 But I think it's probably fine if Fanglin gave the ok. Operations has it's own branches, so it won't interfere w/any of that. You can go ahead and remove tests from rt.conf.

FernandoAndrade-NOAA commented:
This PR is next in queue for testing, please sync and resolve conflicts, thank you.

grantfirl commented:
> This PR is next in queue for testing, please sync and resolve conflicts, thank you.

@FernandoAndrade-NOAA Finished with syncing with develop branches.

gspetro-NOAA commented:
I'm seeing the baseline creation for `hafs_global_multiple_4nests_atm_intel` go DEAD on Derecho after trying to run it twice. My latest run_dir is here: `/glade/derecho/scratch/gpetro/FV3_RT/rt_87539`. Test dir is: `/glade/derecho/scratch/gpetro/RTs/ufs-wm/2810a/tests`. It succeeded on Hercules and Orion. @FernandoAndrade-NOAA you might have to follow up on this with Grant, since I'm out pretty much all day Thursday and Friday. All the other baselines on Derecho have been updated.

FernandoAndrade-NOAA commented:
> I'm seeing the baseline creation for `hafs_global_multiple_4nests_atm_intel` go DEAD on Derecho after trying to run it twice. My latest run_dir is here: `/glade/derecho/scratch/gpetro/FV3_RT/rt_87539`. Test dir is: `/glade/derecho/scratch/gpetro/RTs/ufs-wm/2810a/tests`. It succeeded on Hercules and Orion. @FernandoAndrade-NOAA you might have to follow up on this with Grant, since I'm out pretty much all day Thursday and Friday. All the other baselines on Derecho have been updated.

I'm also seeing a failure on the `hafs_global_multiple_4nests_atm_intel` test for baseline creation after a rerun, `control_wam_intel` has timed out twice as well on Ursa. @grantfirl

FernandoAndrade-NOAA commented:
On the hafs test `err`: `10: FATAL from PE     5: compute_qs: saturation vapor pressure table overflow, nbad= `
Please refer to `/scratch3/NAGAPE/epic/Fernando.Andrade-maldonado/stmp/RT_RUNDIRS/Fernando.Andrade-maldonado/FV3_RT/rt_357822/hafs_global_multiple_4nests_atm_intel` on Ursa

FernandoAndrade-NOAA commented:
Please refer to #1742 and #1907 for a similar issue and potential resolution.

grantfirl commented:
> > I'm seeing the baseline creation for `hafs_global_multiple_4nests_atm_intel` go DEAD on Derecho after trying to run it twice. My latest run_dir is here: `/glade/derecho/scratch/gpetro/FV3_RT/rt_87539`. Test dir is: `/glade/derecho/scratch/gpetro/RTs/ufs-wm/2810a/tests`. It succeeded on Hercules and Orion. @FernandoAndrade-NOAA you might have to follow up on this with Grant, since I'm out pretty much all day Thursday and Friday. All the other baselines on Derecho have been updated.
> 
> I'm also seeing a failure on the `hafs_global_multiple_4nests_atm_intel` test for baseline creation after a rerun, `control_wam_intel` has timed out twice as well on Ursa. @grantfirl

I had 4 tests time out on my latest Ursa run. Is the common practice just to increase wall clock time for tests that repeatedly time out?

I'll look into the hafs_global_multiple_4nests_atm_intel failure. It's odd that my run completed just fine, albeit with differing results. With the instability, the code changes really shouldn't be changing the results more than what one gets with a compiler change, so perhaps this test was already running "on the edge" of instability before?

FernandoAndrade-NOAA commented:
> > > I'm seeing the baseline creation for `hafs_global_multiple_4nests_atm_intel` go DEAD on Derecho after trying to run it twice. My latest run_dir is here: `/glade/derecho/scratch/gpetro/FV3_RT/rt_87539`. Test dir is: `/glade/derecho/scratch/gpetro/RTs/ufs-wm/2810a/tests`. It succeeded on Hercules and Orion. @FernandoAndrade-NOAA you might have to follow up on this with Grant, since I'm out pretty much all day Thursday and Friday. All the other baselines on Derecho have been updated.
> > 
> > 
> > I'm also seeing a failure on the `hafs_global_multiple_4nests_atm_intel` test for baseline creation after a rerun, `control_wam_intel` has timed out twice as well on Ursa. @grantfirl
> 
> I had 4 tests time out on my latest Ursa run. Is the common practice just to increase wall clock time for tests that repeatedly time out?
> 
> I'll look into the hafs_global_multiple_4nests_atm_intel failure. It's odd that my run completed just fine, albeit with differing results. With the instability, the code changes really shouldn't be changing the results more than what one gets with a compiler change, so perhaps this test was already running "on the edge" of instability before?

In general we'd like to avoid just increasing the wall clock for tests that have begun to persistently time out, it may be more appropriate to temporarily disable and assign an issue to it to follow up on. I will retry in the meantime.

gspetro-NOAA commented:
I've also got an error on Hercules for `control_gfs_mpas_gnu`, which starts with: 
```
7: Program received signal SIGSEGV: Segmentation fault - invalid memory reference.
 7:
 7: Backtrace for this error:
 6: ==== backtrace (tid: 186922) ====
 6:  0 0x000000000005f14c ucs_callbackq_cleanup()  ???:0
 6:  1 0x000000000005f30a ucs_callbackq_cleanup()  ???:0
 6:  2 0x0000000000054d90 __GI___sigaction()  :0
 6:  3 0x000000000157340e __gfs_time_vary_pre_MOD_gfs_time_vary_pre_timestep_init()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/2810/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/GFS_time_vary_pre.fv3.F90:122
 6:  4 0x0000000001540c62 __ccpp_mpas_rrfs_time_vary_cap_MOD_mpas_rrfs_time_vary_tsinit_cap()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/stmp/gpetro/FV3_RT/rt_2786501/compile_atm_mpas_dyn32_gnu/build_fv3_atm_mpas_dyn32_gnu/UFSATM/ccpp/physics/ccpp_mpas_rrfs_time_vary_cap.F90:151
 6:  5 0x00000000014a1e34 __ccpp_static_api_MOD_ccpp_physics_timestep_init()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/stmp/gpetro/FV3_RT/rt_2786501/compile_atm_mpas_dyn32_gnu/build_fv3_atm_mpas_dyn32_gnu/UFSATM/ccpp/physics/ccpp_static_api.F90:123
 6:  6 0x000000000149f3b0 __ccpp_driver_MOD_ccpp_step()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/2810/UFSATM/ccpp/CCPP_driver.F90:133
...
```
My most recent run_dir is at `/work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/stmp/gpetro/FV3_RT/rt_2786501`, and my test directory is `/work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/2810/tests`.

All other tests on Hercules are passing.

gspetro-NOAA commented:
Note also that there are increased warnings on Ursa and Orion: https://github.com/ufs-community/ufs-weather-model/actions/runs/23607264996?pr=2810
My understanding is that these will need to be resolved before the PR can be merged. But @NickSzapiro-NOAA feel free to weigh in.

grantfirl commented:
> Note also that there are increased warnings on Ursa and Orion: https://github.com/ufs-community/ufs-weather-model/actions/runs/23607264996?pr=2810 My understanding is that these will need to be resolved before the PR can be merged. But @NickSzapiro-NOAA feel free to weigh in.

@NickSzapiro-NOAA I can see which compilations increase warnings, but since there are already so many CCPP warnings, how are we supposed to know which warnings were added as a result of this PR in order to fix them? For this script to pass, one would just need to fix the number of extra warnings that were reported, but it doesn't matter which warnings are fixed in order to pass this test?

NickSzapiro-NOAA commented:
The idea is to block introducing more warnings so 1) have standard for code quality and 2) don't have burden of fixing them later for implementations.

Unfortunately, I've just been reading the compile/err logs to ID warnings. I guess one way is to diff the compile/err log from this branch and develop (cleanest if done with same path since the entire filepath is in compile/err)

I'll take a look on Ursa, unless someone already knows what the warnings are

grantfirl commented:
> I've also got an error on Hercules for `control_gfs_mpas_gnu`, which starts with:
> 
> ```
> 7: Program received signal SIGSEGV: Segmentation fault - invalid memory reference.
>  7:
>  7: Backtrace for this error:
>  6: ==== backtrace (tid: 186922) ====
>  6:  0 0x000000000005f14c ucs_callbackq_cleanup()  ???:0
>  6:  1 0x000000000005f30a ucs_callbackq_cleanup()  ???:0
>  6:  2 0x0000000000054d90 __GI___sigaction()  :0
>  6:  3 0x000000000157340e __gfs_time_vary_pre_MOD_gfs_time_vary_pre_timestep_init()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/2810/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/GFS_time_vary_pre.fv3.F90:122
>  6:  4 0x0000000001540c62 __ccpp_mpas_rrfs_time_vary_cap_MOD_mpas_rrfs_time_vary_tsinit_cap()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/stmp/gpetro/FV3_RT/rt_2786501/compile_atm_mpas_dyn32_gnu/build_fv3_atm_mpas_dyn32_gnu/UFSATM/ccpp/physics/ccpp_mpas_rrfs_time_vary_cap.F90:151
>  6:  5 0x00000000014a1e34 __ccpp_static_api_MOD_ccpp_physics_timestep_init()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/stmp/gpetro/FV3_RT/rt_2786501/compile_atm_mpas_dyn32_gnu/build_fv3_atm_mpas_dyn32_gnu/UFSATM/ccpp/physics/ccpp_static_api.F90:123
>  6:  6 0x000000000149f3b0 __ccpp_driver_MOD_ccpp_step()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/2810/UFSATM/ccpp/CCPP_driver.F90:133
> ...
> ```
> 
> My most recent run_dir is at `/work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/stmp/gpetro/FV3_RT/rt_2786501`, and my test directory is `/work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/2810/tests`.
> 
> All other tests on Hercules are passing.

This was missed since this test doesn't run on Ursa. I think that I have a fix for it that I'm testing now.

grantfirl commented:
> The idea is to block introducing more warnings so 1) have standard for code quality and 2) don't have burden of fixing them later for implementations.

A worthy goal!

> 
> Unfortunately, I've just been reading the compile/err logs to ID warnings. I guess one way is to diff the compile/err log from this branch and develop (cleanest if done with same path since the entire filepath is in compile/err)
> 
> I'll take a look on Ursa, unless someone already knows what the warnings are

I looked at one of the compile/err logs that were flagged and there were just too many to sort through to try to pick out which ones may have been added. Once warnings are reduced to 0, then this setup would work great as-is. Until then, it would be very helpful for PR originators if the script would do this diff for us to save some human time/work.

NickSzapiro-NOAA commented:
I checked and also quickly sketched out something with Gemini AI ... new warnings look to be cosmetic source line truncated in comments

<details>
  <summary>Click to expand for compiler warnings</summary>

compile_atm_debug_dyn32_intel
compile_atmaq_debug_intel
compile_atml_debug_intel
compile_rrfs_dyn32_phy32_debug_intel
compile_rrfs_dyn64_phy32_debug_intel
compile_s2s_32bit_sfs_debug_intel
compile_s2sw_32bit_pdlib_debug_intel
compile_wam_debug_intel
[Nick.Szapiro@ufe01 uwm_2810]$ for d in uwm_develop/tests/run_dir/compile_*/; do comp_dir=$(basename "$d"); python compare_warnings_text.py ufs-weather-model/tests/run_dir/${comp_dir}/err uwm_develop/tests/run_dir/${comp_dir}/err; done
FAILED: 3 new warning(s) introduced:

/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(496): warning #5194: Source line truncated.
                  !Accumulated tendency = accumulated tendency + current tendency
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(509): warning #5194: Source line truncated.
                  !Current state = current state + dt*(accumulated tendency + current tendency)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(528): warning #5194: Source line truncated.
                  !Current tendency unchanged (but will be overwritten during next primary scheme)
------------------------------------------------------------------------^
------------------------------------------------------------------------
FAILED: 6 new warning(s) introduced:

/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(496): warning #5194: Source line truncated.
                  !Accumulated tendency = accumulated tendency + current tendency
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(509): warning #5194: Source line truncated.
                  !Current state = current state + dt*(accumulated tendency + current tendency)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(528): warning #5194: Source line truncated.
                  !Current tendency unchanged (but will be overwritten during next primary scheme)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_calstrm.f90(30): warning #8889: Explicit interface or EXTERNAL declaration is required.   [CALSTRM]
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_bound.f90(28): warning #8889: Explicit interface or EXTERNAL declaration is required.   [BOUND]
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_calpot.f90(28): warning #8889: Explicit interface or EXTERNAL declaration is required.   [CALPOT]
    call calstrm(Z1D, STRM)
    call BOUND(FLD, FMIN, FMAX)
    call calpot(P1D, T1D, THETA)
---------^
---------^
---------^
------------------------------------------------------------------------
FAILED: 4 new warning(s) introduced:

/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(496): warning #5194: Source line truncated.
                  !Accumulated tendency = accumulated tendency + current tendency
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(509): warning #5194: Source line truncated.
                  !Current state = current state + dt*(accumulated tendency + current tendency)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(528): warning #5194: Source line truncated.
                  !Current tendency unchanged (but will be overwritten during next primary scheme)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_callcl.f90(90): warning #8889: Explicit interface or EXTERNAL declaration is required.   [CALLCL]
------------------------------------------------------------------------
FAILED: 4 new warning(s) introduced:

/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(342): warning #5194: Source line truncated.
          if (istsun(i) > 0) xcosz(i) = xcosz(i) / istsun(i)  ! mean cosine of solar zenith angle at current time
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(496): warning #5194: Source line truncated.
                  !Accumulated tendency = accumulated tendency + current tendency
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(509): warning #5194: Source line truncated.
                  !Current state = current state + dt*(accumulated tendency + current tendency)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(528): warning #5194: Source line truncated.
                  !Current tendency unchanged (but will be overwritten during next primary scheme)
------------------------------------------------------------------------^
------------------------------------------------------------------------
FAILED: 7 new warning(s) introduced:

/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(496): warning #5194: Source line truncated.
                  !Accumulated tendency = accumulated tendency + current tendency
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(509): warning #5194: Source line truncated.
                  !Current state = current state + dt*(accumulated tendency + current tendency)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(528): warning #5194: Source line truncated.
                  !Current tendency unchanged (but will be overwritten during next primary scheme)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Radiation/radiation_gases.f(131): warning #5194: Source line truncated.
      integer, parameter, public :: NF_VGAS = 10   ! number of gas species
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Radiation/radiation_aerosols.f(635): warning #5194: Source line truncated.
------------------------------------------------------------------------^
        if ( iaermdl==0 .or. iaermdl==5 ) then      ! opac-climatology scheme
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Radiation/radiation_gases.f(134): warning #5194: Source line truncated.
      integer, parameter         :: MINYEAR = 1957 ! earlist year 2D CO2 data available
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Radiation/radiation_aerosols.f(643): warning #5194: Source line truncated.
------------------------------------------------------------------------^
        elseif ( iaermdl==1 .or. iaermdl==2 .or. iaermdl==6) then  ! gocart clim/prog scheme
------------------------------------------------------------------------^
------------------------------------------------------------------------
FAILED: 5 new warning(s) introduced:

/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(496): warning #5194: Source line truncated.
                  !Accumulated tendency = accumulated tendency + current tendency
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(509): warning #5194: Source line truncated.
                  !Current state = current state + dt*(accumulated tendency + current tendency)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(528): warning #5194: Source line truncated.
                  !Current tendency unchanged (but will be overwritten during next primary scheme)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_bound.f90(28): warning #8889: Explicit interface or EXTERNAL declaration is required.   [BOUND]
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_dewpoint.f90(30): warning #8889: Explicit interface or EXTERNAL declaration is required.   [DEWPOINT]
    call BOUND(FLD, FMIN, FMAX)
    call DEWPOINT(VP, TD)
---------^
---------^
------------------------------------------------------------------------
FAILED: 5 new warning(s) introduced:

/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(496): warning #5194: Source line truncated.
                  !Accumulated tendency = accumulated tendency + current tendency
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(509): warning #5194: Source line truncated.
                  !Current state = current state + dt*(accumulated tendency + current tendency)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(528): warning #5194: Source line truncated.
                  !Current tendency unchanged (but will be overwritten during next primary scheme)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_calvis.f90(57): warning #8889: Explicit interface or EXTERNAL declaration is required.   [CALVIS]
    call CALVIS(QV, QC, QR, QI, QS, TT, PP, VIS)
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_calpot.f90(28): warning #8889: Explicit interface or EXTERNAL declaration is required.   [CALPOT]
---------^
    call calpot(P1D, T1D, THETA)
---------^
------------------------------------------------------------------------
FAILED: 9 new warning(s) introduced:

/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(496): warning #5194: Source line truncated.
                  !Accumulated tendency = accumulated tendency + current tendency
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(509): warning #5194: Source line truncated.
                  !Current state = current state + dt*(accumulated tendency + current tendency)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/dcyc2t3.f(528): warning #5194: Source line truncated.
                  !Current tendency unchanged (but will be overwritten during next primary scheme)
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Radiation/radiation_aerosols.f(551): warning #5194: Source line truncated.
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/ccpp/physics/physics/Radiation/radiation_aerosols.f(2389): warning #5194: Source line truncated.
        elseif ( iaermdl==1 .or. iaermdl==2 .or. iaermdl==6) then ! use gocart aerosols
------------------------------------------------------------------------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_calpot.f90(28): warning #8889: Explicit interface or EXTERNAL declaration is required.   [CALPOT]
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_calstrm.f90(30): warning #8889: Explicit interface or EXTERNAL declaration is required.   [CALSTRM]
    call calpot(P1D, T1D, THETA)
    call calstrm(Z1D, STRM)
---------^
---------^
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_bound.f90(28): warning #8889: Explicit interface or EXTERNAL declaration is required.   [BOUND]
------------------------------------------------------------------------
/scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810/ufs-weather-model/UFSATM/upp/unit_tests/test_dewpoint.f90(30): warning #8889: Explicit interface or EXTERNAL declaration is required.   [DEWPOINT]
    call BOUND(FLD, FMIN, FMAX)
    call DEWPOINT(VP, TD)
---------^
---------^
</details>

<details>
  <summary> Click for example python script </summary>
import sys
import os
import re
import glob
from collections import defaultdict

def parse_log_blocks(filepath):
    """
    Parses a log file into distinct warning blocks.
    Returns a list of tuples: (normalized_block_string, original_block_string)
    """
    # Matches: /absolute/path/.../ufs-weather-model/Relative/Path.f(123): warning #5194: Message
    #header_regex = re.compile(r'^.*?ufs-weather-model/(.*?)\(\d+\):\s*(warning\s*#\d+:\s*.*)')
    header_regex = re.compile(r'^.*?UFSATM/(.*?)\(\d+\):\s*(warning\s*#\d+:\s*.*)')
    
    blocks = []
    curr_norm = []
    curr_orig = []
    
    try:
        with open(filepath, 'r', encoding='utf-8', errors='ignore') as f:
            for line in f:
                match = header_regex.match(line)
                if match:
                    # Save the previous block if we were tracking one
                    if curr_orig:
                        blocks.append(("".join(curr_norm), "".join(curr_orig)))
                        
                    rel_path = match.group(1).strip()
                    warn_msg = match.group(2).strip()
                    
                    # Start a new block
                    curr_orig = [line]
                    # The normalized version hides the line number so shifted lines cancel out
                    curr_norm = [f"{rel_path}(XXX): {warn_msg}\n"]
                else:
                    if curr_orig:
                        # Intel warning context lines usually start with spaces (code) or dashes (caret)
                        if line.startswith(' ') or line.startswith('-'):
                            curr_orig.append(line)
                            curr_norm.append(line)
                        else:
                            # We hit standard build output, close out this block
                            blocks.append(("".join(curr_norm), "".join(curr_orig)))
                            curr_orig = []
                            curr_norm = []
                            
            # Catch the last block if the file ends abruptly
            if curr_orig:
                blocks.append(("".join(curr_norm), "".join(curr_orig)))
    except Exception as e:
        print(f"Error reading {filepath}: {e}")
        
    return blocks

def parse_target(target_path):
    """Aggregates all warning blocks across a directory or a single file."""
    target_warnings = defaultdict(list)
    
    # Allow the user to pass either a single file or a directory
    if os.path.isfile(target_path):
        log_files = [target_path]
    else:
        log_files = glob.glob(os.path.join(target_path, '*.log')) + \
                    glob.glob(os.path.join(target_path, '*.err')) + \
                    glob.glob(os.path.join(target_path, 'make.out'))
    
    for log_file in log_files:
        blocks = parse_log_blocks(log_file)
        for norm_block, orig_block in blocks:
            target_warnings[norm_block].append(orig_block)
            
    return target_warnings

def main(dev_path, pr_path):
    dev_warnings = parse_target(dev_path)
    pr_warnings = parse_target(pr_path)
    
    total_new_warnings = 0
    report = []
    
    for norm_block, pr_orig_list in pr_warnings.items():
        dev_orig_list = dev_warnings.get(norm_block, [])
        
        # If PR has more occurrences of this exact code block than develop, it's new
        new_count = len(pr_orig_list) - len(dev_orig_list)
        
        if new_count > 0:
            total_new_warnings += new_count
            
            # Print the newly added original blocks (taking the last `new_count` ones)
            for new_orig_block in pr_orig_list[-new_count:]:
                report.append(new_orig_block.strip())
                report.append("-" * 72)

    if total_new_warnings == 0:
        print("Success: No new warnings introduced in this build!")
        sys.exit(0)
    else:
        print(f"FAILED: {total_new_warnings} new warning(s) introduced:\n")
        print("\n".join(report))
        sys.exit(1)

if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: python compare_single_dir.py <dev_compile_dir_or_file> <pr_compile_dir_or_file>")
        sys.exit(1)
    main(sys.argv[1], sys.argv[2])
        
</details>

It has to be cleaned up but the idea is to match warnings but allow for basepath and linenumber changes...then print the PR's compiler warning if no match found. Script used on Ursa at /scratch3/NCEPDEV/nems/Nick.Szapiro/tasks/cm_ufs/uwm_2810

grantfirl commented:
> I checked and also quickly sketched out something with Gemini AI ... new warnings look to be cosmetic source line truncated in comments

@NickSzapiro-NOAA 
This is very helpful, thanks. I fixed the comments that went beyond column 72 in dcyc2t3: https://github.com/ufs-community/ccpp-physics/pull/299/changes/8cc471396e22039b0a4ca0ed484b85c2c3d0ff31.

Some of the other additional warnings from the script reference files that weren't touched by this PR, though, so the only ones that I fixed were in the files that are actually modified in this PR. Since this functionality is using RT logs, I'd rather not try to run all RTs again just to prove that the warnings are fixed. I assume that when the final tests are run again by EPIC, the warnings script will get run again to check on this.

grantfirl commented:
> I've also got an error on Hercules for `control_gfs_mpas_gnu`, which starts with:
> 
> ```
> 7: Program received signal SIGSEGV: Segmentation fault - invalid memory reference.
>  7:
>  7: Backtrace for this error:
>  6: ==== backtrace (tid: 186922) ====
>  6:  0 0x000000000005f14c ucs_callbackq_cleanup()  ???:0
>  6:  1 0x000000000005f30a ucs_callbackq_cleanup()  ???:0
>  6:  2 0x0000000000054d90 __GI___sigaction()  :0
>  6:  3 0x000000000157340e __gfs_time_vary_pre_MOD_gfs_time_vary_pre_timestep_init()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/2810/UFSATM/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/GFS_time_vary_pre.fv3.F90:122
>  6:  4 0x0000000001540c62 __ccpp_mpas_rrfs_time_vary_cap_MOD_mpas_rrfs_time_vary_tsinit_cap()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/stmp/gpetro/FV3_RT/rt_2786501/compile_atm_mpas_dyn32_gnu/build_fv3_atm_mpas_dyn32_gnu/UFSATM/ccpp/physics/ccpp_mpas_rrfs_time_vary_cap.F90:151
>  6:  5 0x00000000014a1e34 __ccpp_static_api_MOD_ccpp_physics_timestep_init()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/stmp/gpetro/FV3_RT/rt_2786501/compile_atm_mpas_dyn32_gnu/build_fv3_atm_mpas_dyn32_gnu/UFSATM/ccpp/physics/ccpp_static_api.F90:123
>  6:  6 0x000000000149f3b0 __ccpp_driver_MOD_ccpp_step()  /work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/2810/UFSATM/ccpp/CCPP_driver.F90:133
> ...
> ```
> 
> My most recent run_dir is at `/work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/stmp/gpetro/FV3_RT/rt_2786501`, and my test directory is `/work2/noaa/epic/gpetro/hercules/RTs/ufs-wm/2810/tests`.
> 
> All other tests on Hercules are passing.

I tried to test a fix on Ursa, but this test runs out of memory on that platform (before and after the fix). I am trying to run on derecho, where I have an account, but it is still in the queue. If someone else can run the `control_gfs_mpas_gnu` test on a supported platform to see if my fix works, that would be very helpful.

NickSzapiro-NOAA commented:
Sorry for my atrocious formatting. Glad it could help.

control_gfs_mpas_gnu passed now on hercules @grantfirl

grantfirl commented:
@gspetro-NOAA @FernandoAndrade-NOAA @BinLiu-NOAA 

I reduced the timestep for the hafs_global_multiple_4nests_atm test from 90 to 60s: https://github.com/ufs-community/ufs-weather-model/pull/2810/commits/b1cbe2dea7c4628aa4334f00a0fd1629c04478f9

This seemed to fix the intermittent instability issue. @BinLiu-NOAA Please let us know if it is not OK to do this for this particular HAFS test.

grantfirl commented:
> Sorry for my atrocious formatting. Glad it could help.
> 
> control_gfs_mpas_gnu passed now on hercules @grantfirl

@NickSzapiro-NOAA Thanks for the help. Compilation has finished for me on Derecho, but still waiting on the run to finish. Perhaps the project code that I'm using is overdrawn or something.

grantfirl commented:
@FernandoAndrade-NOAA This may be ready to be tested again. I think that the last remaining snafu was the control_wam_intel time out. I verified that a WLCK of 45 minutes will work for this if desired. This test already has a special 45 minute WLCLK value for Hera, so perhaps the same for Ursa is warranted.

BrianCurtis-NOAA commented:
What are others opinions to instead reduce FHMAX in tests that fail to meet 30 minute wallclock instead of increasing the wallclock?

DeniseWorthen commented:
@BrianCurtis-NOAA In my opinion, reducing FHMAX should be preferred to extending wall clock. That will change baselines and should be done as a separate Issue and PR. An upcoming PR which changes coupled baselines would be perfect.

gspetro-NOAA commented:
Just noting that my attempt at baseline creation passed on Derecho for `hafs_global_multiple_4nests_atm_intel`, and `control_gfs_mpas_gnu` passes on Hercules with the latest changes.

gspetro-NOAA commented:
I think my remaining questions are: 
- Has @BinLiu-NOAA approved the timestep change from 90 to 60?
- Will there be changes on the `control_wam_intel` test to fix the timeout?

grantfirl commented:
> I think my remaining questions are:
> 
> * Has @BinLiu-NOAA approved the timestep change from 90 to 60?
> * Will there be changes on the `control_wam_intel` test to fix the timeout?

I agree with @DeniseWorthen that fixing the `control_wam_intel` timeout issue by reducing fhmax should be done in a different PR. If we need to make sure new baselines for this PR are created for this test, can we temporarily just adjust WLCLK, and change it back before merging?

Is there a better way to get @BinLiu-NOAA 's attention RE: the HAFS test instability/timestep issue?

dpsarmie commented:
Bin's at the AMS Tropical conference this week, but the HAFS group is working on updating the HAFS RTs so I don't think it's an issue to update that test.

gspetro-NOAA commented:
@grantfirl Could you sync w/develop so tht we can start testing this PR?

grantfirl commented:
> @grantfirl Could you sync w/develop so tht we can start testing this PR?

@gspetro-NOAA Finished.

BrianCurtis-NOAA commented:
I found an issue that needs to be addressed by the Acorn SA's. If its addressed before the final RT Log is pushed, i'll try to get Acorn completed. For now, consider it skippable.

gspetro-NOAA commented:
Testing has completed successfully. Skipping Acorn. I've left a note in sub-PRs.

BrianCurtis-NOAA commented on .gitmodules:
```diff
@@ -18,6 +18,10 @@
+[submodule "HYCOM"]
+  path = HYCOM-interface/HYCOM
+  url = https://github.com/NOAA-EMC/HYCOM-src
+  branch = emc/develop
```
HYCOM shouldn't be here any more.

grantfirl replied:
@BrianCurtis-NOAA Fixed
